test/__snapshots__/fromJSON.test.ts.snap

```diff
@@ -27,19 +27,36 @@ export type IsoDateString = string
 export type RecordIdString = string
 export type HTMLString = string
 
-// System fields
-export type BaseSystemFields<T = never> = {
-	id: RecordIdString
-	collectionId: string
-	collectionName: Collections
-	expand?: T
-}
+// Utility type to check if T is exactly unknown
+type IsExactlyUnknown<T> =
+  unknown extends T
+    ? T extends unknown
+      ? keyof T extends never
+        ? true
+        : false
+      : false
+    : false
```

Comment:
Should this just return the expected type and be joined with `BaseSystemFields`? This would prevent some repetition and be more explicit about what it's doing.

```suggestion
// Utility type to get expand field. If T is provided, expand is no longer optional
type ExpandType<T> =
  unknown extends T
    ? T extends unknown
      ? { expand?: unknown }
      : { expand: T }
    : { expand: T }
```